back/engines/commercial/idea_assignment/app/controllers/idea_assignment/patches/web_api/v1/ideas_controller.rb

```diff
@@ -3,6 +3,15 @@ module Patches
     module WebApi
       module V1
         module IdeasController
+          def idea_attributes
```

Comment:
I've seen conditional attributes being included in the Serializer too, as shown here: https://github.com/Netflix/fast_jsonapi#conditional-attributes

Is there an agreement on when to favor one way over the other?

Reply:
This is not used for serialization, but to list the parameters that are accepted to create or update an idea.

There is another way to implement this though, which is through the policy. Arguably, the policy is a better place to express "who can modify/set which attributes of an idea?". However, it was necessary to move out the logic from the policy in this case, because we need access to the parameters. I think I once read that, by design, pundit does not allow access to the parameters. I need access to the parameters for the case where both the project and the assignee are set (but not yet persisted): the assignee has to be a moderator of the project we're also trying to set.